src/controller/discoverymgr/mnedc/client/client.go

```diff
@@ -348,11 +348,18 @@ func (c *Client) HandleError(err error) {
 
 		c.intf = intf
 		err = c.ParseVirtualIP(string(readBuf[0:n])) //unique TUN ip sent by server
-		if err == nil {
-			err = tunIns.SetTUNIP(c.intf.Name(), c.virtualIP, c.netMask, true)
-			err = tunIns.SetTUNStatus(c.intf.Name(), true, true)
+		if err != nil {
+			log.Println(logPrefix, "Parse Virtual IP error:", err.Error())
+			return
```

Comment:
I'm wondering this is right approach since this code is in the loop and other some [code](https://github.com/lf-edge/edge-home-orchestration-go/blob/7e9ed35d0cbdf46f9975c5bce5c2315201355eca/src/controller/discoverymgr/mnedc/client/client.go#L323) call **continue** instead of return.

Reply:
> 
> 
> I'm wondering this is right approach since this code is in the loop and other some [code](https://github.com/lf-edge/edge-home-orchestration-go/blob/7e9ed35d0cbdf46f9975c5bce5c2315201355eca/src/controller/discoverymgr/mnedc/client/client.go#L323) call **continue** instead of return.

Thanks for a very good question and really needs to be discussed.
My position is as follows: errors should be handled.

1. `err = tunIns.SetTUNIP (c.intf.Name (), c.virtualIP, c.netMask, true)`
does the following: assign an `c.virtualIP` address, `c.netMask` address for the `c.intf.Name ()`interface. 

General example for better understanding:
```
$ ifconfig eth0 172.16.25.125 netmask 255.255.255.22
```

2. `err = tunIns.SetTUNStatus(c.intf.Name(), true, true)`
performs the activation of the `c.intf.Name()` interface.

General example for better understanding:
```
$ ip link set dev eth0 up mtu 1400 qlen 300
```

**Can we activate the interface (command #2) if we have an error setting addresses (command #1)?**
If I misunderstood something or you have other thoughts, please write to me

Reply:
@MoonkiHong @t25kim Of course, if we are considering working in a loop, then we need to think about replacing the following `return` 
```
		intf, err := tunIns.CreateTUN()
		if err != nil {
			log.Println(logPrefix, "TUN error:", err.Error())
			return
		}

		c.intf = intf
		err = c.ParseVirtualIP(string(readBuf[0:n])) //unique TUN ip sent by server
		if err != nil {
			log.Println(logPrefix, "Parse Virtual IP error:", err.Error())
			return
		}

		err = tunIns.SetTUNIP(c.intf.Name(), c.virtualIP, c.netMask, true)
		if err != nil {
			log.Println(logPrefix, "TUN error:", err.Error())
			return
		}

		err = tunIns.SetTUNStatus(c.intf.Name(), true, true)
		if err != nil {
			log.Println(logPrefix, "TUN error:", err.Error())
			return
		}
```
with a continuation with the output of the corresponding error message
example:
```
if err != nil {
			log.Println(logPrefix, <ERROR>", retrying")
			time.Sleep(retryDelay)
			conn.Close()
			continue
		}
```

Reply:
@tdrozdovsky Thank you for your kind analysis. We have also been waiting for the original `MNEDC` proposal developer from @suresh-lc and @sun-sharma. I personally think that we might be able to handle the logic issue in a separated PR linking with the registered (dedicated) issue #196. @tdrozdovsky @suresh-lc @t25kim What do you think?

Reply:
@MoonkiHong yes, I agree with you to allocate to another pool request. Give me your consent and I will remove this code from this pull request and create another one to fix this code.

P.S. It would be nice for us to close alerts from the LGTM faster!

Reply:
@tdrozdovsky This PR is to resolve the reported LGTM issues. Let us merge this PR once all the reviewers approve this. Then let us create a separate PR to handle the logic issue as you requested. As we suggested, it would be better for us to have every PR with a single feature/enhancement from https://github.com/lf-edge/edge-home-orchestration-go/blob/master/.github/CONTRIBUTING.md (Section 5).

Reply:
@MoonkiHong Currently, we have 17 alerts. We will not be able to fix all with one pull request, since they are of different types. And I think that we will fix them step by step as there are more complex that will be considered for a long time, and also some that require some modification before fixing them. 

And I agree with your comment: "We have also been waiting for the original MNEDC proposal developer from @suresh-lc and @sun-sharma"

Reply:
> @MoonkiHong Currently, we have 17 alerts. We will not be able to fix all with one pull request, since they are of different types. And I think that we will fix them step by step as there are more complex that will be considered for a long time, and also some that require some modification before fixing them.
> 
> And I agree with your comment: "We have also been waiting for the original MNEDC proposal developer from @suresh-lc and @sun-sharma"

@tdrozdovsky Seems that you agree with my proposal. Let us go ahead step-by-step. ^__^

Reply:
@tdrozdovsky Thank you for sharing your idea with us. I know your code has no problem at the moment. I left the comment just for the future work.